serve/mlc_serve/engine/engine_common.py

```diff
@@ -229,26 +229,46 @@ def get_requests_to_process(
     return requests, is_prompt_batch, token_counts
 
 
-def should_stop_by_length(state: RequestState, max_context_length: int) -> bool:
+def should_stop_seq_by_length(
+    gen_seq: GenerationSequence,
+    prompt_len: int,
+    max_context_length: int,
+    max_tokens: Optional[int],
+) -> bool:
     # TODO: currently, we simply return true for both stopping reasons.
     #       in the future, we can differentiate these two.
     # this include prompt tokens and gen tokens so far
```

Comment:
@sunggg @elvin-n I don't know who wrote this comment originally, but is this still relevant? I don't understand what it is saying

Reply:
I wrote this comment. I wondered if it would be useful if we return more information whether the generation is stopped because of max context length or max token. It is more like `NOTE` rather than `TODO`, but now that I think of this, we can remove this.